toolchain/check/eval.cpp

```diff
@@ -109,6 +117,17 @@ static auto GetConstantValue(Context& context, SemIR::InstBlockId inst_block_id,
   return context.inst_blocks().Add(const_insts);
 }
 
+// The constant value of a type block is that type block, but we still need to
+// extract its phase.
+static auto GetConstantValue(Context& context, SemIR::TypeBlockId type_block_id,
+                             Phase* phase) -> SemIR::TypeBlockId {
+  auto types = context.type_blocks().Get(type_block_id);
+  for (auto type_id : types) {
+    GetConstantValue(context, type_id, phase);
```

Comment:
Should this short-circuit if the phase is runtime?

Reply:
That's not possible: we only form a`TypeId`s for type constants, so we can't find a runtime subexpression here. That does mean that we could bail out if we hit an error, but given that it's a bit subtle why that would be correct, and it would only improve performance along an error path (and marginally slow down the other paths), maybe it's not worth it.